scalr/resource_scalr_provider_configuration.go

```diff
@@ -0,0 +1,487 @@
+				Optional:     true,
+				MaxItems:     1,
+				ExactlyOneOf: []string{"aws", "azurerm", "custom"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"project": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"credentials": {
+							Type:      schema.TypeString,
+							Optional:  true,
+							Sensitive: true,
+						},
+					},
+				},
+			},
+			"azurerm": {
+				Type:         schema.TypeList,
+				Optional:     true,
+				MaxItems:     1,
+				ExactlyOneOf: []string{"aws", "google", "custom"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"client_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"client_secret": {
+							Type:      schema.TypeString,
+							Optional:  true,
+							Sensitive: true,
+						},
+						"subscription_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"tenant_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"custom": {
+				Type:         schema.TypeList,
+				Optional:     true,
+				MaxItems:     1,
+				ExactlyOneOf: []string{"aws", "google", "azurerm"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"provider_name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"argument": {
+							Type:     schema.TypeSet,
+							Required: true,
+							MinItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"value": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"sensitive": {
+										Type:     schema.TypeBool,
+										Optional: true,
+										Default:  false,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceScalrProviderConfigurationCreate(d *schema.ResourceData, meta interface{}) error {
+	scalrClient := meta.(*scalr.Client)
+
+	name := d.Get("name").(string)
+	accountID := d.Get("account_id").(string)
+
+	configurationOptions := scalr.ProviderConfigurationCreateOptions{
+		Name:                 scalr.String(name),
+		Account:              &scalr.Account{ID: accountID},
+		ExportShellVariables: scalr.Bool(d.Get("export_shell_variables").(bool)),
+	}
+	var createArgumentOptions []scalr.ProviderConfigurationParameterCreateOptions
+
+	if v, ok := d.GetOk("aws"); ok {
+		configurationOptions.ProviderName = scalr.String("aws")
+
+		aws := v.([]interface{})[0].(map[string]interface{})
+		if access_key, ok := aws["access_key"].(string); ok {
+			configurationOptions.AwsAccessKey = scalr.String(access_key)
+		}
+		if secret_key, ok := aws["secret_key"].(string); ok {
+			configurationOptions.AwsSecretKey = scalr.String(secret_key)
+		}
+
+	} else if v, ok := d.GetOk("google"); ok {
```

Comment:
@DayS1eeper check please if we clear option if we change provider from gce to aws

Reply:
@penja provider can't be changed, `provider_name` is not updatable in API, also error is raised on scale provider side if `provider_name` (e.g.  `gce` -> `aws`) is changed
https://github.com/Scalr/terraform-provider-scalr/pull/127/files#diff-e7c0b438989841ddb4feebb24352000b23756131f98d780821ff92619a91ce8eR29